lib/topology_lib_sflowtool/parser.py

```diff
@@ -78,49 +77,169 @@ def parse_sflowtool(raw_output):
+                assert len(sflow_packet) == len(flow_packet_fields)
+                for field in range(len(sflow_packet)):
+                    packet[flow_packet_fields[field]] = sflow_packet[field]
+                flow_count = flow_count + 1
+                packets.append(packet)
+            elif sflow_packet[0] == 'CNTR':
+                assert len(sflow_packet) == len(sample_packet_fields)
+                for field in range(len(sflow_packet)):
+                    packet[sample_packet_fields[field]] = sflow_packet[field]
+                sample_count = sample_count + 1
+                packets.append(packet)
+
+        result['flow_count'] = flow_count
+        result['sample_count'] = sample_count
+        result['packets'] = packets
+        return result
+
+    elif mode == 'detail':
+
+        start_datagram = 'startDatagram =================================\n'
+        end_datagram = 'endDatagram   =================================\n'
+
+        start_sample = 'startSample ----------------------\n'
+        end_sample = 'endSample   ----------------------\n'
+
+        finder = r'{}(.*?){}'
+
+        datagrams = findall(
+            finder.format(start_datagram, end_datagram), raw_output, DOTALL)
+
+        result = []
+        attribute_re = '((?!startSample|endSample)(.+?) (.+))'
+        start_sample_re = 'startSample .+'
+        end_sample_re = 'endSample .+'
+
+        for datagram in datagrams:
+
+            result.append(OrderedDict())
+            datagram_lines = datagram.splitlines()
+            sample_flag = False
+
+            for datagram_line in datagram_lines:
+
+                attribute = match(attribute_re, datagram_line)
+                start_sample = match(start_sample_re, datagram_line)
+                end_sample = match(end_sample_re, datagram_line)
+
+                if attribute is not None:
+                    if sample_flag:
+                        result[-1]['samples'][-1][attribute.group(2)] = \
+                            attribute.group(3)
+                        continue
+                    else:
+                        result[-1][attribute.group(2)] = attribute.group(3)
+                        continue
+
+                elif start_sample is not None:
+                    if sample_flag:
+                        raise Exception('Nested sample found.')
+                    sample_flag = True
+
+                    if 'samples' not in result[-1].keys():
+                        result[-1]['samples'] = [OrderedDict()]
+                        continue
+                    else:
+                        result[-1]['samples'].append(OrderedDict())
+
+                elif end_sample is not None:
+                    if not sample_flag:
+                        raise Exception(
+                            'Ending of sample found without \
+                            it being started before.'
+                        )
+                    sample_flag = False
+
```

Comment:
Please add statistics such as -
1. Total number of datagrams
2. Total FLOW samples
3. Total CNTR samples